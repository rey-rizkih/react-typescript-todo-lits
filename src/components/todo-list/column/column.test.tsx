import { shallow } from "enzyme";
import TodoColumn, { TodoColumnProps } from ".";

const mockTodoColumnProps: TodoColumnProps = {
  column: {
    id: "active",
    title: "Active",
    rowsId: ["active-1", "active-2"],
  },
};

jest.mock("react-beautiful-dnd", () => ({
  Droppable: ({ children }) =>
    children(
      {
        draggableProps: {
          style: {},
        },
        provided: {
          innerRef: jest.fn(),
        },
      },
      {}
    ),
  Draggable: ({ children }) =>
    children(
      {
        draggableProps: {
          style: {},
        },
        innerRef: jest.fn(),
      },
      {}
    ),
  DragDropContext: ({ children }) => children,
}));

let wrapper;

describe("<TodoColumn/> rendering", () => {
  beforeEach(() => {
    wrapper = shallow(<TodoColumn {...mockTodoColumnProps} />);
  });

  // it("renders correctly", () => {
  //   expect(wrapper).toMatchSnapshot();
  // });

  it("should render one", () => {
    expect(wrapper.find("Droppable")).toHaveLength(1);
  });

  it("should have an children", () => {
    expect(
      wrapper.find("Droppable").renderProp("children")()
    ).toHaveBeenCalled();
  });

  // it("should render <TodoRow/> for each todo", () => {
  //   expect(wrapper.find("TodoRow")).toHaveLength(
  //     mockTodoColumnProps.column.rowsId.length
  //   );
  // });
});
