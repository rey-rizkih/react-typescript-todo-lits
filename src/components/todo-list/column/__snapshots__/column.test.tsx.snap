// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoColumn/> rendering renders correctly 1`] = `
<Connect(Droppable)
  direction="vertical"
  droppableId="active"
  getContainerForClone={[Function]}
  ignoreContainerClipping={false}
  isCombineEnabled={false}
  isDropDisabled={false}
  mode="standard"
  renderClone={null}
  type="TASK"
>
  <Component />
</Connect(Droppable)>
`;
